src/options.js

```diff
@@ -39,15 +41,18 @@ export function getRealmOptions({
   debugNames = false,
   uniqueSuffix,
   timeout,
+  residual,
+  serialize,
```

Comment:
You could specify this as `residual = false, serialize = !residual` and may encode a default in both are explicitly specified as false. E.g. throw an error. That way the options and any defaults are fully resolved inside of this module.